### PR TITLE
Document Library

### PR DESCRIPTION
**Pull request by lwjameson**
This PR adds a Document Library which allows you to choose organizational documents or to upload new documents to the library.

* In order to to not confuse things, documents already in your project are not shown.
* You can select multiple documents to add.

### CODE REVIEW

File: src/apps/project-home/ProjectHome.tsx [hunk 1]
```diff
@@ -44,184 +66,249 @@ export const ProjectHome = (props: ProjectHomeProps) => {
...
-        <UploadTracker 
+        <div style={isDragActive ? { pointerEvents: 'none' } : undefined}>
+          <DocumentLibrary
+            open={addOpen}
+            i18n={props.i18n}
+            onAddDocument={onDocumentSelected}
+            onCancel={() => setAddOpen(false)}
+            user={props.user}
+            dataDirty={dataDirty || documentUpdated}
+            clearDirtyFlag={() => {
```
Comment (rsimon) [inline src/apps/project-home/ProjectHome.tsx:278]: Will the `dataDirty` flag trigger a full refetch of all documents in Supabase whenever the user makes a change? If so: might not be an issue immediately, but would we have to think about a lighter approach eventually? (In other cases, I updated state locally adding/replacing/deleting only single items.)

**Review by rsimon — CHANGES_REQUESTED** [review]

**Comment by rsimon** [comment]
Hi @lwjameson, I think this is good to go, with only minor polishing.

My only real question is about the dirty flag: how that works (see comment above), and whether it might eventually lead to performance issues further down the line. (Unless I misunderstood what it does!) Either way: IMO doesn't need to be addressed now. Just something to keep in mind for later.

File: src/themes/default/dialog/index.css [hunk 2]
```diff
@@ -62,3 +61,19 @@
   display: flex;
   margin-top: 2.5em;
 }
+
+.dialog-content-doc-lib {
```
[the remark this answers is not in the document]
Comment (lwjameson) [reply src/themes/default/dialog/index.css]: Done!

File: src/pages/[lang]/projects/[project]/index.astro [hunk 3]
```diff
@@ -4,46 +4,62 @@ import { createSupabaseServerClient } from '@backend/supabaseServerClient';
...
-if (documents.error) { 
+if (documents.error) {
   const path = Astro.url.protocol + '//' + Astro.url.host;
   const error = await fetch(`${path}/500`);
-  return new Response(error.body, { headers: error.headers, status: 500 }); 
+  return new Response(error.body, { headers: error.headers, status: 500 });
+}
+
+//const orgDocuments = [];
+const orgDocuments = await listAllDocuments(supabase);
```
[the remark this answers is not in the document]
Comment (lwjameson) [reply src/pages/[lang]/projects/[project]/index.astro]: Yes. This should be removed. Done!

File: src/backend/helpers/documentHelpers.ts [hunk 4]
```diff
@@ -3,22 +3,34 @@ import { createDocument } from '@backend/crud';
...
+  Document,
+  DocumentInContext,
+  DocumentInTaggedContext,
+  Layer,
+  TaggedContext,
+} from 'src/Types';
 import { getTagsForContext } from './tagHelpers';
+import { supabase } from '@backend/supabaseBrowserClient';
+
+export type DbResult<T> = T extends PromiseLike<infer U> ? U : never;
```
[the remark this answers is not in the document]
Comment (lwjameson) [reply src/backend/helpers/documentHelpers.ts]: Eh. Some WIP I guess I moved on from. Removed.

**Pull request merged**